Sources/SwiftProtobuf/JSONDecodingError.swift

```diff
@@ -44,9 +44,9 @@ public enum JSONDecodingError: Error {
     /// We hit the end of the JSON string and expected something more...
     case truncated
     /// A JSON Duration could not be parsed
-    case malformedDuration
```

Comment:
We can't remove the old cases as it will break someones code that was trying to catch that.  Go ahead and tag it as deprecated with a note about the newer error.

@tbkka Since this is a rename of an error code, it is "breaking", but I'm guessing most folks just catch the errors generically, and don't inspect them.  Are you ok with change change or would you rather hold it for a major version bump because of the api change?

Reply:
There are a number of cases in this enum that are named in the same way, in the style of `malformedType`. It seems odd to rename just two of them and only on them provide this diagnostic information on those.

I also clearly don't understand the language spec well enough to know why it permits you to duplicate the case name with associated data using tuple types that are compatible. It seems very strange. They look like functions with named parameters but in the grammar the associated data is clearly a tuple. I am not even sure how I would expect pattern matching to work against these new cases?

Reply:
Tuple labels are part of the type. That is, `(x: Int)` and `(y: Int)` are distinct types. Some literals can be used to refer to either, but they are different types nonetheless.

Pattern matching would work like so:

```swift
switch error {
case .malformed(timestamp: let timestamp):
    break
case .malformed(duration: let duration):
    break
```

If you wanted to pattern match on just the case name, that is no longer possible, you'll need this:

```
case .malformed(timestamp: _):
    break
case .malformed(duration: _):
    break
```

This is definitely an API break. However, so is adding new fields, and you've also allowed that in the past (e.g. 3dec79b4dc2ed931b5a167f2f81354e63b5f5c0c). How it's handled is kind of up to the project. NIO would forbid this change, if you wanted the context: we'd require a new error enum and move to throwing those.